joomla/joomla-cms
# com_users Logout Menu Item Type

This Pull Requests adds a new Menu Item Type "Logout" to the available menu item types. It allows users to create a menu item "Logout" that directly logouts a user when clicking on the menu item. Optionally you can set a page to redirect after logout, otherwise the user will be redirected to the default (home) page. 

In this way it is no longer needed to use the "Login" menu-item type to logout, which shows a page with a "Logout" button only. 

For the `JFIELD_LOGOUT_REDIRECT_PAGE_DESC` and `JFIELD_LOGOUT_REDIRECT_PAGE_LABEL` the same description is used as for the mod_login with the similar functionality (`MOD_LOGIN_FIELD_LOGOUT_REDIRECTURL_DESC` and `MOD_LOGIN_FIELD_LOGOUT_REDIRECTURL_LABEL`).
# Test instructions

After applying the patch you can create a new menu-item from type "Logout"
<img width="666" alt="screen shot 2015-08-16 at 20 24 48" src="https://cloud.githubusercontent.com/assets/522834/9294677/df843032-4454-11e5-9829-845f1dbbf2fb.png">

Provide a title and typically you would set the Access to "Registered" so only logged-in users can see the logout menu-item. 

Optionally you can select a "Logout Redirection Page" under "Options" that will be used to redirect the user to after logout. "Default" will return to the default (home) page.
<img width="676" alt="screen shot 2015-08-16 at 20 29 18" src="https://cloud.githubusercontent.com/assets/522834/9294707/7f3825a2-4455-11e5-878b-6463bdadda78.png">

Save the menu item and login with a user on the frontend of the website. The logout menu item should be visible. Clicking on the logout menu item should end your user session and redirect to the page set.
